core/src/main/scala/org/apache/spark/storage/TachyonBlockManager.scala

```diff
@@ -95,8 +94,23 @@ private[spark] class TachyonBlockManager() extends ExternalBlockManager with Log
   override def putBytes(blockId: BlockId, bytes: ByteBuffer): Unit = {
     val file = getFile(blockId)
     val os = file.getOutStream(WriteType.TRY_CACHE)
-    os.write(bytes.array())
-    os.close()
+    try {
+      os.write(bytes.array())
+      os.close()
```

Comment:
There is a code path where neither is cancelled though -- any subclass of `RuntimeException` or `Error` will invoke neither `cancel()` nor `close()`. If `close()` is safely idempotent then it should be in a `finally` block to catch all cases.